src/resources/candlepin-internationalization.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash -x
+
+evalrc() {
+    if [ "$1" -ne "0" ]; then
+        echo "$2"
+        exit $1
+    fi
+}
+
+# This function converts candlepin version to a int value with (Zero) padding
+# which helps in comparing versions.
+# Examples -
+# 2.5.6  -> 2005006000
+# 3.2.0  -> 3002000000
+# 2.9.6  -> 2009006000
+get_simplified_version() {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+sudo git fetch --all
+
+for GIT_BRANCH in master candlepin-3.1-HOTFIX candlepin-2.9-HOTFIX
+do
+
+  #Clean the project
+  sudo ./gradlew clean
+  evalrc $? "Project cleaning was not successful for branch $GIT_BRANCH."
+
+  #Checkout the branch
+  sudo git checkout $GIT_BRANCH
+  evalrc $? "Checkout branch : $GIT_BRANCH was not successful."
+
+  # Check if we need to use Java 11, otherwise auto detect Java version.
+  # Candlepin support Java 11 from 3.2.0 & onwards.
+
+  CANDLEPIN_BASE_VERSION=$(grep "Version:" server/candlepin.spec.tmpl | awk -F"Version: " '{print $2}')
+  JAVA_11_SUPPORTED_VERSION=3.2.0
+
+  echo "Base version of candlepin: " $CANDLEPIN_BASE_VERSION
+  if [ $(get_simplified_version $CANDLEPIN_BASE_VERSION) -ge $(get_simplified_version $JAVA_11_SUPPORTED_VERSION) ] ; then
+      JAVA_VERSION=11
+      echo "Using Java 11 for branch $GIT_BRANCH."
+      sudo update-alternatives --set java /usr/lib/jvm/java-$JAVA_VERSION-openjdk-$JAVA_VERSION*/bin/java
+  else
+      JAVA_VERSION=1.8.0
+      echo "Using Java 8 for branch $GIT_BRANCH."
+      sudo update-alternatives --set java /usr/lib/jvm/java-$JAVA_VERSION-openjdk-$JAVA_VERSION*/jre/bin/java
+  fi
+
+  #To generate the auto generated classes and files
+  sudo ./gradlew war
+  evalrc $? "Compilation was not successful for branch $GIT_BRANCH."
+
+  #Execute the gettext task
+  sudo ./gradlew gettext
+  evalrc $? "Gettext was not successful for branch $GIT_BRANCH."
+
+  #Executing the msgattrib task
+  sudo ./gradlew msgattrib
+  evalrc $? "msgattrib was not successful for branch $GIT_BRANCH."
+
+  #Code to commit the updated files
```

Comment:
One more important thing I forgot: There is a chance (in fact, I'd say that will be most times this job will run, especially for the HOTFIX branches), that there will be no actual translation strings being removed at all, and therefore there will be no commit needed.

Unfortunately, just running `./gradlew gettext`, will alter the common/po/keys.pot file by altering the `POT-Creation-Date:` in that file, even though no other changes were made. in that case, we don't want to git add/commit/push at all (no commits should be done for that branch), so we need a way to detect this at this point in the script.

The nanny-script actually does perform this check by running `git diff | egrep -v -e '^( |\+#|\-#|@@|\+\+\+|\-\-\-|diff|index)' -e 'X-Generator' -e 'POT-Creation-Date'`. If this returns nothing, then nothing significant has changed, so we shouldn't commit/push anything.

Reply:
Updated.